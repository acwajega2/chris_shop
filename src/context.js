import React,{Component} from 'react';




//--------Creating the context....
const ProductContext = React.createContext();


//-----------------Creating the Provider Class-----
class ProductProvider extends Component{
    constructor(){
        super();
        this.state ={
            numCount :0
        }
    }
   
    //------------Methods to run when the componet has mounted'
    componentDidMount(){

        console.log('Greate, I did Mount to the DOM....');
    }
    //--------------------End of mounting functions----
  addNumbers =(a) =>{
      return a+2
  }

    //------------Increasing the count---------
increaseCount = () =>{

    let initiaCount = this.state.numCount;

    this.setState(() =>{
        return {
            numCount : initiaCount+ 1
        }
    })

}
//-----------------End of Count increment


    render() {
        return(
            <ProductContext.Provider  value={{...this.state,increaseCount :this.increaseCount,addNumbers :this.addNumbers}}>
                {this.props.children}
            </ProductContext.Provider>
        )
    }

}
//--------------End of the Provider Class


const ProductConsumer = ProductContext.Consumer;

export {ProductProvider,ProductConsumer}
