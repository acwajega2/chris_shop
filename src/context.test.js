import React from 'react';
import renderer from 'react-test-renderer';
import {shallow,mount} from './enzyme';


import {ProductProvider} from './context';



describe("MAIN CONTEXT", ()=>{

      //------------------Snap Shot Testing-------
  it("Matches the snap Shot", () =>{
    const tree = renderer.create(<ProductProvider />).toJSON();
    expect(tree).toMatchSnapshot();

});
//------------------End of Snap Shot Testing--------


//----------Should increment the values
it("Should perform an increment",() =>{
     const component = mount(<ProductProvider />);
console.log(component)



})

    //----------Checking if it initializes properly
    it("Should Initialize", ()=>{
        shallow(<ProductProvider />);
    })
})