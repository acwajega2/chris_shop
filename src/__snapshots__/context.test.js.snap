exports[`MAIN CONTEXT Matches the snap Shot 1`] = `null`;
